### DISCUSSION TITLE
PostDraw Callback using CompositeViewer

### DISCUSSION THREAD

**Discussion by SayagAnat** [openscenegraph/OpenSceneGraph#1286] [Q&A] [answered]
Hi, I am trying to create a CompositeViewer applications which contains 4 views on the same scene. I created MyCallback Class which inherits from osg::Camera::DrawCallback and implements operator ().
In main:
MyCallback* pMyCallBack = new MyCallback();
CompositeViewer viewer;
...
osgViewer::View myView= CreateView(...);
viewer,addView(myView);
myView->getCamera()->setFinalDrawCallBack(pMyCallBack);
...

For some reason pMyCallBack is not invoked.
What can be a reason for that?

**Comment by robertosfield** [comment, marked as answer]
There are too many unknowns for use to be able to give an answer - only you have the complete source code to your own code.  Yu also have access to the whole source code of the OSG so you are the one best placed to directly investigate what might be going on.

The best I can do is recommend that your build the OSG and your own code in debug and then step through the viewer draw traversal code to see where the various camera level callbacks are invoked and see what it's doing/not doing, this should give up pointers to why your own code isn't being invoked as you'd expect.
